### PR TITLE
Test views fix n1 problems

### PR DESCRIPTION
**Pull request by tiagomarin**
### :heavy_check_mark: -  Use Capybara to write integration tests for each view in your project.
  
- User index page:
  - I can see the username of all other users.
  - I can see the profile picture for each user.
  - I can see the number of posts each user has written.
  - When I click on a user, I am redirected to that user's show page.
  
- user show page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see the user's bio.
  - I can see the user's first 3 posts.
  - I can see a button that lets me view all of a user's posts.
  - When I click a user's post, it redirects me to that post's show page.
  - When I click to see all posts, it redirects me to the user's post's index page.

- User post index page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see a post's title.
  - I can see some of the post's body.
  - I can see the first comments on a post.
  - I can see how many comments a post has.
  - I can see how many likes a post has.
  - I can see a section for pagination if there are more posts than fit on the view.
  - When I click on a post, it redirects me to that post's show page.

- Post show page:
  - I can see the post's title.
  - I can see who wrote the post.
  - I can see how many comments it has.
  - I can see how many likes it has.
  - I can see the post body.
  - I can see the username of each commentor.
  - I can see the comment each commentor left.
### :speech_balloon: Feel free to shoot me a message in the pull request if you have any questions.

Thank you for your time!

### CODE REVIEW

File: spec/views/users/index.html.erb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  it 'loads user names correctly into the page' do
+    expect(page).to have_content('Tom', count: 1)
+    # same as => expect(page).to have_css("h3", text: "Tom")
+    expect(page).to have_css('img', count: 1)
+    expect(page).to have_content("nº of posts: #{@user.posts.length}")
+  end
+  # it "redirects to user/1/posts when clicked on 'Show all posts' button" do
+  #   click_link "Ana" # too many users with same name
+  #   expect(current_path).to eql(user_path(User.first))
+  # end
```
Comment (AshakaE) [inline spec/views/users/index.html.erb_spec.rb]: Kindly add the following tests as it is a requirement for `USER Index Page` 

- [x] Tests that redirects to the user's show page when you click on a user.
- [x] Tests that asserts you can see the number of posts each user has written.
- [x] Tests that asserts you can see the profile picture for each user.
- [x] Tests that asserts you can see the username of all other users.

File: spec/views/posts/index.html.erb_spec.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  end
+
+  it 'renders comments of post' do
+    expect(page.find("#comments-post-#{@post.id}").all('li').length).to eql(@post.comments.length)
+  end
+
+  it "redirects to user/1/post/x when clicked on a post'" do
+    page.find("#post-#{@post.id}").click
+    expect(current_path).to eql(user_post_path(@user, @post))
+  end
```
Comment (AshakaE) [inline spec/views/posts/index.html.erb_spec.rb]: Kindly add the following tests for `User Post Index Page`

- [ ] Tests that asserts you can see a section for pagination if there are more posts than fit on the view.
- [x] Tests that asserts you can see how many likes a post has.
- [x] Tests that redirects to the post's show page when you click on a post.

**Review by AshakaE — CHANGES_REQUESTED** [review]
Hi @tiagomarin , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Git flow ✅ 
- Pull request description 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/users/index.html.erb_spec.rb (hunk 1 above)
Comment (tiagomarin) [reply spec/views/users/index.html.erb_spec.rb]: Hi @AshakaE,

Thanks for reviewing my code :+1: 
I don't know why I commented that part that checks the redirection... and the rest is there, I'll organize better. :heavy_check_mark:

File: spec/views/posts/index.html.erb_spec.rb (hunk 2 above)
Comment (tiagomarin) [reply spec/views/posts/index.html.erb_spec.rb]: @AshakaE 

Could you clarify a little more about first point?

"Tests that asserts you can see a section for pagination if there are more posts than fit on the view."

All my posts show up in the view, as this is users/show path, it is supposed to show only the latest 3. I'm kinda confused, could you explain a little further?

**Review by m05e5 — APPROVED** [review]

**Review by m05e5 — APPROVED** [review]
Hi @tiagomarin , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
goodjob

## Highlights
✔️ your linters are all ok
✔️ Descriptive PR 

## Optional suggestions


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — COMMENTED** [review]
Hi @tiagomarin , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has been merged, please unmerge it as this allows for a review to be done

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by Munsa1 — COMMENTED** [review]
Hi @tiagomarin, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is merged, merged Pr''s do not have an option for a reviewer to fully review your project, that is why we are instructed to invalidate such projects. kindly unmerge your Pr and resubmit the link, so a review can be made. Thank you
Follow these simple [steps](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/reverting-a-pull-request)

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by youmari — COMMENTED** [review]
# Project Approved :trophy: :tada: 🟢 

Hi @tiagomarin 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linters error.
:heavy_check_mark: Details PR title and description.
:heavy_check_mark: Good commit messages.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U02FTTCCU2V)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**